### PR TITLE
feat(rate limits): set default rate limits and caching in Mender Enterprise

### PR DESCRIPTION
**Pull request by tranchitella**
Add a redis pod and configure deviceauth and useradm to connect to it
enabling rate limits and caching of JWT tokens. At the same time, set
default rate limits in tenantadm.

Ticket: MEN-5270

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Review by MaciejTe — APPROVED** [review]

File: mender/templates/api-gateway-configmap.yaml [hunk 1]
```diff
@@ -473,6 +495,15 @@ data:
...
           compress: true
 
+        ratelimit:
+          ratelimit:
+            average: {{ .Values.api_gateway.ratelimit.average }}
+            burst: {{ .Values.api_gateway.ratelimit.burst }}
+            period: {{ .Values.api_gateway.ratelimit.period }}
+            sourcecriterion:
+              ipstrategy:
+                depth: 1
```
Comment (alfrunes) [inline mender/templates/api-gateway-configmap.yaml:505]: Doesn't this parameter depend on the infrastructure you're running Mender on-prem? Should we make it configurable?

**Review by alfrunes — APPROVED** [review]
Thank you for taking care of this :pray:
The PR looks good, I only have one nitpicky question.

File: mender/templates/api-gateway-configmap.yaml [hunk 2]
```diff
@@ -473,6 +495,10 @@ data:
         compression:
           compress: true
 
+        ratelimit:
+          ratelimit:
```
Comment (alfrunes) [inline mender/templates/api-gateway-configmap.yaml:499]: I believe this is case sensitive:
```suggestion
          rateLimit:
```

Comment (tranchitella) [reply mender/templates/api-gateway-configmap.yaml:499]: Fixed, it was a copy and paste from `saas`.
https://github.com/mendersoftware/saas/pull/672

**Pull request merged**